packages/mgt/src/graph/graph.groups.ts

```diff
@@ -64,6 +101,23 @@ export async function findGroups(
 ): Promise<Group[]> {
   const scopes = 'Group.Read.All';
 
+  let cache: CacheStore<CacheGroups>;
+  const key = query || '*' + groupTypes;
+
+  if (groupsCacheEnabled()) {
+    cache = CacheService.getCache(cacheSchema, 'groupsQuery');
+    const groups = await cache.getValue(key);
+    if (groups && getGroupsInvalidationTime() > Date.now() - groups.timeCached) {
+      if (groups.numResults === top) {
+        return groups.groups.map(x => JSON.parse(x));
+      } else if (groups.numResults > top) {
+        // if request is less than the cache's requests, return a slice of the results
+        return groups.groups.map(x => JSON.parse(x)).slice(0, top + 1);
+      }
```

Comment:
Minor optimization by avoiding else if since slice will return all items anyway when top === numResults

```suggestion
      if (groups.numResults >== top) {
        // if request is less than the cache's requests, return a slice of the results
        return groups.groups.map(x => JSON.parse(x)).slice(0, top + 1);
      }
```